mm-go-irckit/userbridge.go

```diff
@@ -49,6 +54,19 @@ func NewUserBridge(c net.Conn, srv Server, cfg *viper.Viper) *User {
 	u.msgCounter = make(map[string]int)
 	u.updateCounter = make(map[string]time.Time)
 
+	statePath := u.v.GetString("mattermost.lastviewedsavepath")
+	if statePath != "" {
+		staleDuration := u.v.GetString("mattermost.lastviewedstaleduration")
+		lastViewedAt, err := loadLastViewedState(statePath, staleDuration)
+		if err == nil {
+			logger.Info("Loaded lastViewedAt from ", time.Unix(lastViewedAt["__LastViewedStateSavedTime__"]/1000, 0))
+			u.lastViewedAt = lastViewedAt
+		} else {
+			logger.Warning("Unable to load saved lastViewedAt, using empty values: ", err)
+		}
```

Comment:
A nitpick, but I like the err!=nil first, as it's more idiomatic
```suggestion
		if err != nil {
			logger.Warning("Unable to load saved lastViewedAt, using empty values: ", err)
		} else {
			logger.Info("Loaded lastViewedAt from ", time.Unix(lastViewedAt["__LastViewedStateSavedTime__"]/1000, 0))
			u.lastViewedAt = lastViewedAt
		}
```

I think it's also better to put line 57 => line 66 in a separate function